edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/authentication/AuthAgentProtocolHead.cs

```diff
@@ -100,8 +100,13 @@ public void Dispose()
             {
                 // FIXES a bug when the the process is stuck on WebHost.StopAsync() method call https://github.com/dotnet/extensions/issues/1363
                 // lets not wait forever but cancel after a timeout
-                var cts = new CancellationTokenSource(TimeSpan.FromSeconds(15));
-                this.CloseAsync(cts.Token).Wait();
+                var timeout = TimeSpan.FromSeconds(15);
+                var closingTask = this.CloseAsync(new CancellationTokenSource(timeout).Token);
```

Comment:
if `CloseAsync` calls `IWebHost.StopAsync` and blocks everything, how does this change helps? or the problem was in blocking call `.Wait()`?

Reply:
Yes, the problem is with blocking `Wait`. On a machine, with a limited number of cores, it was quite easy to block all threads in a thread pool. That's how I reproduced the issue: run tests in parallel on a VM with 2 cores

Reply:
With this change, we request stop and if it hasn't finished within some interval (15 sec), stop waiting and continue. Hopefully, GC will clean up resources soon: when the test finished or when our container restarted in production

Reply:
@dmolokanov, I wonder what would happen on a 1 core machine. `.Result` call on line 106 is also blocking, and isn't it possible that it can deadlock itself so even timer will not be able to wake up? I'm not sure about dotnet, but we recently had the same problem with @and-rewsmith and tokio blocking calls.